### DISCUSSION TITLE
Feature: Wake On Lan

### DISCUSSION THREAD

**Discussion by mechanicalfluff** [open-webui/open-webui#3526] [Ideas]
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
Add the ability for openwebui to send the machine running ollama a WOL. In my case, the ollama server is on a different pc than what is running this container.

**Describe alternatives you've considered**
I wake the ollama server with other apps. I use moonlight on my phone to wake the pc, then close it to use ollama.

**Comment by tjbck** [comment]
I believe this should be doable with the newly introduced Functions feature, keep us updated on your journey!

**Reply by thenchev** [reply]
Hey @tjbck 
I'm also trying to setup WOL. 
I searched for this Functions feature just found this unfinished page: https://docs.openwebui.com/tutorial/functions
I also found the functions in the UI but I just have an empty list of options.
Could you point me to some resources how to set it up?

**Comment by Mavyre** [comment]
Okay so I'm having the exact same case. The GPU server is used occasionnaly, so I need it to be off most of the time. For another project using the GPU server, I already used a Docker WOL Api (https://github.com/rix1337/WakeOnLAN-API).

I needed to WOL the GPU server whenever needed from the open-webui interface. Functions might not be working, because when Ollama is not available, all models disappear from the user interface. I can't simply write a filter where the inlet functions wakes on lan, wait for Ollama availability, and let the query go through, because no models are displaying and user can't query any model at all.

It's dirty, by I simply added a new Ollama server `http://my-wol-api-host:18080/wol/01:23:45:67:89:ab?` (the `?` at the end is added to ignore anything added by Open Webui). Whenever someone asks for models, it wakes up the server. And that just... Works. Sometimes, people need to refresh the webui once and the models appear.

Even plugging it to Brave's Leo or Continue.dev wakes it up on demand.

**Reply by gtinjr** [reply]
I am new to open-webui and I am interested in doing something similar.
 Did you end up writing a function that calls your wol service? Would you be able to share the function code or expand your answer on the approach you used?
Thanks

**Reply by Mavyre** [reply]
Nope, no function as when the Ollama server is offline, the models disappear from OpenWebui interface. Thus, functions won't be able to be executed.
As I stated on my answer, I used a Docker WOL API that wakes up the server whenever it receives a GET/POST request. You can read more there: https://github.com/rix1337/WakeOnLAN-API

After that, add the wakeup url as an Ollama API Connections in Open Webui, and your server will wakeup whenever you access Open Webui (or you use Open Webui chat API, which is very useful for Léo/Continue.dev.

**Reply by gtinjr** [reply]
I endup going a different route that should work for any service using the ollama endpoint.  I setup a spring boot cloud gateway setvice running on my kubernetes k3s sbc cluster.  The gateway will take ollama requests at domain gateway.example.net/ollama. Ollama service runs on the desktop computer at ollama.example.net:11434. When the gateway gets a request and the server(my desktop computer)  is not reachable then it sends a wake on lan magic packet and it waits for the server to be reachable before forwarding the request. Dns is handled by pihole running on a pi zero connected to the usb port of my router.

**Reply by Mavyre** [reply]
Great, a wake-on-lan reverse proxy!

**Reply by HunterAhlquist** [reply]
> Nope, no function as when the Ollama server is offline, the models disappear from OpenWebui interface. Thus, functions won't be able to be executed. As I stated on my answer, I used a Docker WOL API that wakes up the server whenever it receives a GET/POST request. You can read more there: https://github.com/rix1337/WakeOnLAN-API
> 
> After that, add the wakeup url as an Ollama API Connections in Open Webui, and your server will wakeup whenever you access Open Webui (or you use Open Webui chat API, which is very useful for Léo/Continue.dev.

I have an issue with this solution, Open WebUI hits the /api/tags subdirectory thus not triggering WOL. How did you bypass this?

**Reply by Mavyre** [reply]
Just add a ? At the end of the URL. /api/tags is passed as a parameter instead of an actual path. Works hehe.
Also, I enhanced my wake on Lan by writing a pipe model instead of that solution. It works great 😃

**Comment by mdlmarkham** [comment]
I'm using WoL through Home Assistant - is there a consensus on the best way to do this through OpenWebUI directly?

**Comment by borislemer** [comment]
Hi guys,

I have the open-webui frontend server and the ollama/lms in the backend (sleeping ;) ) , a service was created to sniff for traffic direction inference and if the inference server is down, it gets a WOL packet.

Please use as desired @open-webui would be cool if you integrate smthg like this:

https://github.com/borislemer/wake_wol

Regards,
Boris

**Comment by monksy** [comment]
You can do this with a proxy. 

I did this and documented it here: https://theexceptioncatcher.com/2025/07/remotely-accessing-your-local-ai-services-with-wake-on-lan-a-web-proxy/

This even works with voice mode because you can do an SSL proxy.

**Reply by a-priestley** [reply]
> You can do this with a proxy.
> 
> I did this and documented it here: https://theexceptioncatcher.com/2025/07/remotely-accessing-your-local-ai-services-with-wake-on-lan-a-web-proxy/
> 
> This even works with voice mode because you can do an SSL proxy.

This is so close to being exactly what I need, but there is no obvious way to add request headers for connections in open webui.
Would you mind sharing your specific configuration?
Edit:
I now realize there is an easy way to add request headers, but only for openAI API type connections [(Commit)](https://github.com/open-webui/open-webui/commit/f25b7b73b4d4f7a3d60f79091797a10fe9a7f530).
Perhaps this work can be easily reused for Ollama type connections as well.
